### ISSUE TITLE
Unhandled error in deflate compression handling

### ISSUE THREAD

**Issue by pocketjoso** [bitinn/node-fetch#45]
Okay, I have to admit that I have no idea what's going on here, sorry for the vague issue.

Node `v0.12.7`
```
var fetch = require('node-fetch')
fetch('http://time-attendance.co.uk')
```

gives
```
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: incorrect header check
    at Zlib._handle.onerror (zlib.js:366:17)
```

...
The fetch results in a `ok` response... The promise is not rejected. But I still get this error thrown,  somewhere.. Confused. :/

**Comment by pocketjoso** [comment]
Ah. Just curled that url. A _redirect_? Not supported in this library?

**Comment by pocketjoso** [comment]
How do I handle this error? And is it possible to follow the redirect using `fetch-node`?

**Comment by bitinn** [comment]
Yep, `node-fetch` follows redirect by default, so the problem is not the redirect itself, but somehow node.js zlib is not able to handle the `deflate` response this server is returning.

https://github.com/bitinn/node-fetch/blob/master/index.js#L141

As we do test for deflate response support, my guess would be that particular server is returning something weird.

https://github.com/bitinn/node-fetch/blob/master/test/server.js#L67-L74
https://github.com/bitinn/node-fetch/blob/master/test/test.js#L329-L338

As a workaround for now, setting options `compress: false` will cause the server to return decompressed content, allowing your fetch to go through.

Neverthless, I think in this case we need to handle this error so that `fetch` reject the result.

**Comment by pocketjoso** [comment]
Thanks for your reply.

>  I think in this case we need to handle this error so that fetch reject the result

Yeah I think so too. As it is now I seem to have no way to handle the error, and I would prefer to throw here than having to change the default compression (I'm making many requests).

> to truly fix the problem require checking what went wrong during zlib inflate, maybe open a ticket on node.js core?

Not very read up on these things. How would I be able to tell what wen't wrong during the `inflate`? Would I not need to get onto the server serving that url?

**Comment by bitinn** [comment]
@pocketjoso I plan to work on this module next week, will update when the fix has landed.

**Comment by pocketjoso** [comment]
Awesome. :)

**Comment by bitinn** [comment]
Fixed in v1.3.3, just note that we can't reject the fetch request (because the error is on body, which is a stream) but we do reject it when you call `res.text()` or `res.json()`.

Otherwise you can use `body.on('error', handler)` to it independently.

**Comment by pocketjoso** [comment]
Excellent, thanks for fixing!
